core/src/test/scala/unit/kafka/server/AbstractFetcherManagerTest.scala

```diff
@@ -210,4 +217,103 @@ class AbstractFetcherManagerTest {
     verify(fetcher).maybeUpdateTopicIds(Set(tp1), topicIds)
     verify(fetcher).maybeUpdateTopicIds(Set(tp2), topicIds)
   }
+
+  @Test
+  def testExpandThreadPool(): Unit = {
+    testResizeThreadPool(10, 50, 6)
+  }
+
+  @Test
+  def testShrinkThreadPool(): Unit = {
+    testResizeThreadPool(50, 10, 6)
+  }
+
+  def testResizeThreadPool(currentFetcherNum: Int, newFetcherNum: Int, brokerNum: Int): Unit = {
```

Comment:
nit: `currentFetcherSize` and `newFetcherSize` would be better names. `brokerNum` is constant, should we define it inline?